system/Commands/Generators/ModelGenerator.php

```diff
@@ -0,0 +1,137 @@
+<?php
+
+/**
+ * This file is part of the CodeIgniter 4 framework.
+ *
+ * (c) CodeIgniter Foundation <admin@codeigniter.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace CodeIgniter\Commands\Generators;
+
+use CodeIgniter\CLI\BaseCommand;
+use CodeIgniter\CLI\CLI;
+use CodeIgniter\CLI\GeneratorTrait;
+
+/**
+ * Generates a skeleton Model file.
+ */
+class ModelGenerator extends BaseCommand
+{
+	use GeneratorTrait;
+
+	/**
+	 * The Command's Group
+	 *
+	 * @var string
+	 */
+	protected $group = 'Generators';
+
+	/**
+	 * The Command's Name
+	 *
+	 * @var string
+	 */
+	protected $name = 'make:model';
+
+	/**
+	 * The Command's Description
+	 *
+	 * @var string
+	 */
+	protected $description = 'Generates a new model file.';
+
+	/**
+	 * The Command's Usage
+	 *
+	 * @var string
+	 */
+	protected $usage = 'make:model <name> [options]';
+
+	/**
+	 * The Command's Arguments
+	 *
+	 * @var array
+	 */
+	protected $arguments = [
+		'name' => 'The model class name.',
+	];
+
+	/**
+	 * The Command's Options
+	 *
+	 * @var array
+	 */
+	protected $options = [
+		'--table'     => 'Supply a table name.',
+		'--dbgroup'   => 'Database group to use. Defaults: "default".',
+		'--return'    => 'Return type, Options: [array, object, entity], Defaults: "array".',
+		'--namespace' => 'Set root namespace. Defaults: "APP_NAMESPACE".',
+		'--suffix'    => 'Append the component title to the class name (e.g. UserModel => User).',
+		'--force'     => 'Force overwrite existing file.',
+	];
+
+	/**
+	 * Actually execute a command.
+	 *
+	 * @param array $params
+	 */
+	public function run(array $params)
+	{
+		$this->component = 'Model';
+		$this->directory = 'Models';
+		$this->template  = 'model.tpl.php';
+
+		$this->execute($params);
+	}
+
+	/**
+	 * Prepare options and do the necessary replacements.
+	 *
+	 * @param string $class
+	 *
+	 * @return string
+	 */
+	protected function prepare(string $class): string
+	{
+		$table   = $this->getOption('table');
+		$DBGroup = $this->getOption('dbgroup') ?? 'default';
+		$return  = $this->getOption('return') ?? 'array';
+
+		if (! is_string($table))
+		{
+			$table = CLI::prompt(lang('CLI.generator.tableName'), null, 'required'); // @codeCoverageIgnore
```

Comment:
@mostafakhudair This is where it stops as it expects an input, like paul suspected.

Reply:
and how to avoid testing this? does codeCoverageIgnore isn't enough?

Reply:
`codeCoverageIgnore ` only tells the coverage script that the marked line should not count as "untested".
If you want to stay that way, you should update the tests such that the `make:model` command always gets passed a table name directly.

Reply:
Maybe you could also add an `echo` command to the tests to simulate user input, but I'm not familiar with that.
Maybe @paulbalandan knows more?

Reply:
You need to add the default table name to all calls to `make:model` in `ModelGeneratorTest.php`.
Maybe try running the tests locally before pushing your changes here.
You can use `phpunit --no-coverage --filter ModelGeneratorTest`.
You can likely ignore all failures that get displayed (at least on Windows) but you will see whether it gets stuck or not.

Reply:
Thank you